src/index.ts

```diff
@@ -1507,11 +1495,10 @@ type BaseResourceId =
 	| {
 			'@': string;
 	  };
-type ResourceId =
-	| BaseResourceId
-	| {
-			[key: string]: BaseResourceId;
-	  };
+type ResourceNaturalKey = {
```

Comment:
Can you call this "alternate key" instead to match the odata spec: https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#_Toc31360936 - it just makes life easier when trying to correlate things here against the spec

Reply:
TIL
Updated @Page-  :+1: